src/EA.Weee.Web/Areas/AeReturn/Controllers/ReturnsController.cs

```diff
@@ -0,0 +1,157 @@
+    using EA.Weee.Web.Services;
+    using EA.Weee.Web.Services.Caching;
+    using EA.Weee.Web.ViewModels.Returns;
+    using System;
+    using System.Threading.Tasks;
+    using System.Web.Mvc;
+
+    public class ReturnsController : AeReturnBaseController
+    {
+        private readonly Func<IWeeeClient> apiClient;
+        private readonly BreadcrumbService breadcrumb;
+        private readonly IWeeeCache cache;
+        private readonly IMapper mapper;
+
+        public ReturnsController(Func<IWeeeClient> apiClient, BreadcrumbService breadcrumb, IWeeeCache cache, IMapper mapper)
+        {
+            this.apiClient = apiClient;
+            this.breadcrumb = breadcrumb;
+            this.cache = cache;
+            this.mapper = mapper;
+        }
+
+        [HttpGet]
+        public virtual async Task<ActionResult> Index(Guid organisationId)
+        {
+            using (var client = apiClient())
+            {
+                var @returns = await client.SendAsync(User.GetAccessToken(), new GetReturns(organisationId, FacilityType.Ae));
+
+                var viewModel = mapper.Map<ReturnsViewModel>(@returns);
+
+                viewModel.OrganisationName = (await client.SendAsync(User.GetAccessToken(), new GetOrganisationInfo(organisationId))).OrganisationName;
+                viewModel.OrganisationId = organisationId;
+
+                await SetBreadcrumb(organisationId, BreadCrumbConstant.AeReturn);
+
+                return View(viewModel);
+            }
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public ActionResult Index(ReturnsViewModel viewModel)
+        {
+            return AeRedirect.ExportedWholeWeee(viewModel.OrganisationId);
+        }
+
+        [HttpGet]
+        public async Task<ActionResult> ExportedWholeWeee(Guid organisationId, Guid? returnId = null)
+        {
+            await SetBreadcrumb(organisationId, BreadCrumbConstant.AeReturn);
+
+            using (var client = apiClient())
+            {
+                if (returnId == null)
+                {
+                    AddReturn request = new AddReturn()
+                    {
+                        OrganisationId = organisationId,
+                        FacilityType = FacilityType.Ae
+                    };
+
+                    returnId = await client.SendAsync(User.GetAccessToken(), request);
+                }
+
+                ExportedWholeWeeeViewModel model = new ExportedWholeWeeeViewModel()
+                {
+                    ReturnId = returnId.GetValueOrDefault()
+                };
+
+                return View(model);
+            }
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<ActionResult> ExportedWholeWeee(Guid organisationId, ExportedWholeWeeeViewModel viewModel)
+        {
+            if (!ModelState.IsValid)
+            {
+                await SetBreadcrumb(organisationId, BreadCrumbConstant.AeReturn);
```

Comment:
Should this be moved out of the if statement? If there's an issue we'll lose the breadcrumb